src/mlpack/methods/ann/layer/fast_lstm.hpp

```diff
@@ -0,0 +1,304 @@
+/**
+ * @file fast_lstm.hpp
+ * @author Marcus Edel
+ *
+ * Definition of the Fast LSTM class, which implements a Fast LSTM network
+ * layer.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_FAST_LSTM_HPP
+#define MLPACK_METHODS_ANN_LAYER_FAST_LSTM_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <limits>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * An implementation of a faster version of the Fast LSTM network layer.
+ * Basically by combining the calculation of the input, forget, output gates
+ * and hidden state in a single step. The standard formula changes as follows:
+ *
+ * @f{eqnarray}{
+ * i &=& sigmoid(W \cdot x + W \cdot h + b)
+ * f &=& sigmoid(W  \cdot x + W \cdot h + b)
+ * z &=& tanh(W \cdot x + W \cdot h + b)
+ * c &=& f \cdot c + i \cdot z
+ * o &=& sigmoid(W \cdot x + W \cdot h + b)
+ * h &=& o \cdot tanh(c)
+ * @f}
+ *
+ * Note that FastLSTM network layer does not use peephole connections between
+ * the cell and gates.
+ *
+ * For more information, see the following.
+ *
+ * @code
+ * @article{Hochreiter1997,
+ *   author  = {Hochreiter, Sepp and Schmidhuber, J\"{u}rgen},
+ *   title   = {Long Short-term Memory},
+ *   journal = {Neural Comput.},
+ *   year    = {1997}
+ * }
+ * @endcode
+ *
```

Comment:
I think it might be worthwhile in this documentation to add a link to the regular `LSTM` class (and in the `LSTM` class, add a link to this class as something quicker that could be used).